packages/react/src/components/provider.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState, useMemo, useEffect, type ReactNode } from 'react';
+import { PenumbraProviderNotAvailableError, PenumbraSymbol } from '@penumbra-zone/client';
+import { penumbraContext } from '../hooks/penumbra';
+
+export const PenumbraProvider = ({
+  children,
+  providerOrigin,
+  requestApproval = false,
+  required = false,
+}: {
+  children?: ReactNode | ReactNode[];
+  providerOrigin: string;
+  requestApproval?: boolean;
+  required?: boolean;
```

Comment:
thought: i don't think the `required` property is needed here. If users want to, they can throw errors by themselves. And generally throwing errors within components is not a good practice